src/engine/Settings.cs

```diff
@@ -174,6 +175,22 @@ private Settings()
     /// </summary>
     public SettingValue<bool> CheatsEnabled { get; set; } = new SettingValue<bool>(false);
 
+    /// <summary>
+    ///   If false username will be set to System username
+    /// </summary>
+    public SettingValue<bool> CustomUsernameEnabled { get; set; } = new SettingValue<bool>(false);
+
+    /// <summary>
+    ///   Username that the user can choose
+    /// </summary>
+    public SettingValue<string> CustomUsername { get; set; } = new SettingValue<string>(null);
+
+    public string CustomUsernameValue =>
```

Comment:
This needs to be JsonIgnore, also a better name would be `ActiveUsername` to indicate that this isn't actually just the value of custom username.